lib/vector-common/src/request_metadata.rs

```diff
@@ -0,0 +1,93 @@
+/// Metadata for batch requests.
+#[derive(Clone, Debug, Default)]
+pub struct RequestMetadata {
+    /// Number of events represented by this batch request.
+    event_count: usize,
+    /// Size, in bytes, of the in-memory representation of all events in this batch request.
+    events_byte_size: usize,
+    /// Size, in bytes, of the estimated JSON-encoded representation of all events in this batch request.
+    events_estimated_json_encoded_byte_size: usize,
+    /// Uncompressed size, in bytes, of the encoded events in this batch request.
+    request_encoded_size: usize,
+    /// On-the-wire size, in bytes, of the batch request itself after compression, etc.
+    ///
+    /// This is akin to the bytes sent/received over the network, regardless of whether or not compression was used.
+    request_wire_size: usize,
+}
+
+// TODO: Make this struct the object which emits the actual internal telemetry i.e. events sent, bytes sent, etc.
+impl RequestMetadata {
+    #[must_use]
+    pub fn new(
+        event_count: usize,
+        events_byte_size: usize,
+        request_encoded_size: usize,
+        request_wire_size: usize,
+        events_estimated_json_encoded_byte_size: usize,
+    ) -> Self {
+        Self {
+            event_count,
+            events_byte_size,
+            events_estimated_json_encoded_byte_size,
+            request_encoded_size,
+            request_wire_size,
+        }
+    }
```

Comment:
>  The biggest reason to keep this is if we want to keep the values private and so read-only through getters.

That was the only reason I added this method actually.
The getters (and the `RequestMetadata` struct itself) were originally implemented before my changes, and my changes moved them to this file and added extra stuff to it.
So the existing struct fields on `RequestMetadata` were already not pub.
Rather than making them pub I chose to add this method but it is only used in two places.

I can remove it and change those fields to pub though if its :ok_hand:  ?

Reply:
Felt like a defeat of the purpose of the getters though :shrug: